lib/tokenize.cpp

```diff
@@ -10040,8 +10036,15 @@ void Tokenizer::simplifyBitfields()
             }
         } else if (Token::Match(typeTok, "%type% : %num%|%bool% ;") &&
                    typeTok->str() != "default") {
-            tok->deleteNext(4 + tokDistance(tok, typeTok) - 1);
-            goback = true;
+            const std::size_t id = anonymousBitfieldCounter++;
+            const std::string name = "__cppcheck_anon_bit_field_" + std::to_string(id) + "__";
+            Token *newTok = typeTok->insertToken(name);
+            newTok->isAnonymousBitfield(true);
+            if (newTok->tokAt(2)->isBoolean())
+                newTok->setBits(static_cast<char>(newTok->strAt(2) == "true"));
+            else
+                newTok->setBits(static_cast<char>(MathLib::toBigNumber(newTok->tokAt(2))));
```

Comment:
I am not sure if setBits should take a 8-bit integer. casts can be avoided if it's a Mathlib::bigint. and it wouldn't hurt with a check that ensures the range..